Add sapphire docs (#171)
oasisprotocol/docs: docs/general/manage-tokens/how-to-transfer-rose-into-evm-paratime.mdx

import DocCard from '@theme/DocCard';
import {findSidebarItem} from '@site/src/sidebarUtils';

# How to Transfer ROSE into an EVM ParaTime

> 

Reviewer: I would add a short summary at the beginning, something like:

> This guide will teach you how to transfer your ROSE tokens into one of our EVM-compatible ParaTimes: Emerald and Sapphire.

Author: And then someone will have to edit this doc to add their EVM-compatible ParaTime. I'd prefer not to hardcode names or falsely imply that first-party PTs have special treatment.

## About

> An EVM-compatible ParaTime is one that integrates the Oasis Runtime SDK's EVM module.

Reviewer: Could you squeeze in something that would relate EVM to Ethereum?

These are end-user docs so they should assume no prior knowledge and I believe Ethereum is more commonly known than EVM.

Author: These are dev docs. If you end up on a JS bundler page but don't know what JS is, are you really in the right place? I'll add a clause about it, but I'd prefer to keep these docs concise rather than maintain content better found elsewhere.

> Currently there are two such ParaTimes: [Emerald](../developer-resources/emerald-paratime/README.mdx) and [Sapphire](../developer-resources/sapphire-paratime/README.mdx)

Reviewer: Add missing `.` at the end.

> Currently there are two such ParaTimes: [Emerald](../developer-resources/emerald-paratime/README.mdx) and [Sapphire](../developer-resources/sapphire-paratime/README.mdx)

Author: ```suggestion
Currently there are two such ParaTimes: [Emerald](../developer-resources/emerald-paratime/README.mdx) and [Sapphire](../developer-resources/sapphire-paratime/README.mdx).
```

An EVM-compatible ParaTime hosts EVM accounts in its own independent chain state, which means

> that ROSE held at the consensus layer is not the same as the EVM paratime's token. Fortunately,

Reviewer: ```suggestion
that ROSE held at the consensus layer are not the same as the EVM paratime's tokens. Fortunately,
```

Author: Nice idea, but I'm referring to the token as an entity. Would you say "Bro, I swapped all of my Eths into USTs and now they went to zero 😭 ?" Probably not, right?

you can easily swap between the two by making _deposits_ and _withdrawals_.
These procedures are described in detail in [ADR 0003](/adrs/0003-consensus-runtime-token-transfer).

> The _deposit_ procedure will first approve the ParaTime to access your specified amount of ROSE. Then, a `Deposit` transaction will be executed inside the ParaTime and tokens will be transferred from your consensus layer account to the ParaTime. If that transaction succeeds, the ParaTime will update the state of your provided EVM account to reflect the deposit of tokens. Emerald and Sapphire will both give you 1 e/sROSE for each consensus ROSE, although other ParaTimes may implement different rules.

Reviewer: ```suggestion
The _deposit_ procedure will first approve a ParaTime to access your specified amount of ROSE. Then, a `Deposit` transaction will be executed inside the ParaTime and tokens will be transferred from your consensus layer account to the ParaTime. If that transaction succeeds, the ParaTime will update the state of your provided EVM account to reflect the deposit of tokens. Emerald and Sapphire will both give you 1 e/sROSE for each consensus ROSE, although other ParaTimes may implement different rules.
```

Author: I'm referring to the paratime that's being targeted. It doesn't randomly choose "a" paratime, it chooses the one you intended to choose.

> The _deposit_ procedure will first approve the ParaTime to access your specified amount of ROSE. Then, a `Deposit` transaction will be executed inside the ParaTime and tokens will be transferred from your consensus layer account to the ParaTime. If that transaction succeeds, the ParaTime will update the state of your provided EVM account to reflect the deposit of tokens. Emerald and Sapphire will both give you 1 e/sROSE for each consensus ROSE, although other ParaTimes may implement different rules.

Reviewer: Huh, `e/sROSE` sounds too complex to me. Could you just write it out, e.g.:

> Emerald and Sapphire will both give you one "Emerald" ROSE / "Sapphire" ROSE for each consensus ROSE, although other ParaTimes may implement different rules.

Author: Prof. Song suggested it. You can take that up with her.
